### DISCUSSION TITLE
Automaticaly Pan on drag

### DISCUSSION THREAD

**Discussion by ericuldall** [bcakmakoglu/vue-flow#573] [Q&A] [answered]
Is there a way to automatically pan the canvas area if you were to drag an element outside of the visible bounds?

**Comment by bcakmakoglu** [comment, marked as answer]
This is not a built-in feature atm, no.
You'd have to do this yourself by starting to to change the viewport position (using `setViewport`) when a node is being dragged towards the edge of the viewport (you can get the dimensions from the store as well, they're saved as `dimensions`).

**Reply by ericuldall** [reply]
Thanks for the insight. IMO, this would be a great feature out of the box, maybe `<VueFlow autoPan />`

**Reply by bcakmakoglu** [reply]
Currently there's a couple other features that have been requested that I have to prioritized but I'll keep this in mind for sure ^^
